R/bookmark-state-local.R

```diff
@@ -0,0 +1,30 @@
+# Function wrappers for saving and restoring state to/from disk when running
+# Shiny locally.
+#
+# These functions provide a directory to the callback function.
+#
+# @param id A session ID to save.
+# @param callback A callback function that saves state to or restores state from
+#   a directory. It must take one argument, \code{stateDir}, which is a
+#   directory to which it writes/reads.
+
+saveInterfaceLocal <- function(id, callback) {
+  # Try to save in app directory, or, if that's not available, in the current
+  # directory.
+  appDir <- getShinyOption("appConfig")$appDir %OR% getwd()
```

Comment:
The case where `appDir` is `NULL` is `shinyApp(ui, server)`, right? Perhaps capture `appDir <- getwd()` during the `shinyApp` call rather than late-binding here?